lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -651,20 +657,23 @@ LogicalResult FirRegOp::canonicalize(FirRegOp op, PatternRewriter &rewriter) {
 OpFoldResult FirRegOp::fold(FoldAdaptor adaptor) {
   // If the register has a symbol or preset value, we can't optimize it away.
   // TODO: Handle a preset value.
-  if (getInnerSymAttr() || getPresetAttr())
+  if (getInnerSymAttr())
     return {};
 
+  auto presetAttr = getPresetAttr();
+
   // If the register is held in permanent reset, replace it with its reset
   // value. This works trivially if the reset is asynchronous and therefore
   // level-sensitive, in which case it will always immediately assume the reset
   // value in silicon. If it is synchronous, the register value is undefined
   // until the first clock edge at which point it becomes the reset value, in
   // which case we simply define the initial value to already be the reset
-  // value.
-  if (auto reset = getReset())
-    if (auto constOp = reset.getDefiningOp<hw::ConstantOp>())
-      if (constOp.getValue().isOne())
-        return getResetValue();
+  // value. Works only if no preset.
+  if (!presetAttr)
+    if (auto reset = getReset())
+      if (auto constOp = reset.getDefiningOp<hw::ConstantOp>())
+        if (constOp.getValue().isOne())
+          return getResetValue();
```

Comment:
Not directly related but I feel this canonicalization is not correct either... I think reset value needs to constant

Reply:
okay, seemed fine to me, according to the comments in the code. But lets investigate this further, if its legal or not.